pkg/urbit/vere/io/ames.c

```diff
@@ -162,110 +171,102 @@ _ames_alloc(uv_handle_t* had_u,
 static void
 _ames_pact_free(u3_pact* pac_u)
 {
-  if ( 0 == pac_u->typ_y ) {          // ames packet
-  } else if ( 1 == pac_u->typ_y ) {  // fine request
-    c3_free(pac_u->req_u.pat_c);
-  } else {                         // fine response
-    c3_free(pac_u->res_u.pat_c);
-    c3_free(pac_u->res_u.dat_y);
+  switch ( pac_u->typ_y ) {
+    case PACT_AMES:
+      break;
+
+    case PACT_WAIL:
+      c3_free(pac_u->wal_u.ken_u.pat_c);
+      break;
+
+    case PACT_PURR:
+      c3_free(pac_u->pur_u.ken_u.pat_c);
+      c3_free(pac_u->pur_u.mew_u.dat_y);
+      break;
+
+    default:
+      u3l_log("ames_pact_free: bad packet type %d\n", pac_u->typ_y);
+      u3_pier_bail(u3_king_stub());
   }
 
-  c3_free(pac_u->dns_c);
+  c3_free(pac_u->rut_u.dns_c);
   c3_free(pac_u->hun_y);
-
   c3_free(pac_u);
 }
 
 /* _ames_panc_free(): remove references, lose refcounts and free struct
 */
 static void
-_ames_panc_free(u3_panc* pac_u)
+_ames_panc_free(u3_panc* pan_u)
 {
-  if ( 0 != pac_u->nex_u ) {
-    pac_u->nex_u->pre_u = pac_u->pre_u;
+  if ( 0 != pan_u->nex_u ) {
+    pan_u->nex_u->pre_u = pan_u->pre_u;
   }
 
-  if ( 0 != pac_u->pre_u ) {
-    pac_u->pre_u->nex_u = pac_u->nex_u;
+  if ( 0 != pan_u->pre_u ) {
+    pan_u->pre_u->nex_u = pan_u->nex_u;
   }
   else {
-    c3_assert(pac_u == pac_u->sam_u->pac_u);
-    pac_u->sam_u->pac_u = pac_u->nex_u;
+    c3_assert(pan_u == pan_u->pac_u.sam_u->pan_u);
+    pan_u->pac_u.sam_u->pan_u = pan_u->nex_u;
   }
 
-  c3_free(pac_u->ptr_v);
-  c3_free(pac_u);
+  _ames_pact_free(&pan_u->pac_u);
+  c3_free(pan_u);
 }
 
-/* _lane_scry_path(): format scry path for retrieving a lane
-*/
-static inline u3_noun _lane_scry_path(u3_noun who)
+static c3_y
+_ames_prel_size(u3_head* hed_u)
 {
-  return u3nq(u3i_string("peers"),
-              u3dc("scot", 'p', who),
-              u3i_string("forward-lane"),
-              u3_nul);
+  c3_y hed_y = 4;
+  c3_y lif_y = 1;
+  c3_y sen_y = 2 << hed_u->sac_y;
+  c3_y rec_y = 2 << hed_u->rac_y;
+  c3_y rog_y = ( c3y == hed_u->rel_o )? 6 : 0;
+  return hed_y + lif_y + sen_y + rec_y + rog_y;
 }
 
-static void _log_prel(u3_prel* pre_u)
+static c3_s
+_ames_body_size(u3_body* bod_u)
 {
-  u3l_log("-- PRELUDE --\n");
-  u3l_log("sender life: %u\n", pre_u->sic_y);
-  u3l_log("receiver life: %u\n", pre_u->ric_y);
-  u3l_log("sender: %" PRIu64 "\n", pre_u->sen_d[0]);
-  u3l_log("receiver: %" PRIu64" \n", pre_u->rec_d[0]);
-  u3l_log("\n");
+  return bod_u->con_s;
 }
 
-static void _log_requ(u3_requ* req_u)
+static c3_s
+_fine_keen_size(u3_keen* ken_u)
 {
-  _log_prel(&req_u->pre_u);
-  u3l_log("--- REQUEST ---\n");
-  u3l_log("strlen: %u\n", req_u->len_s);
-  u3l_log("path: %s\n", req_u->pat_c);
-  u3l_log("frag: %u\n", req_u->fra_w);
-  u3l_log("\n");
-  return;
+  return 4 + 2 + ken_u->len_s;
```

Comment:
We could consider putting a comment here detailing what the 4 and 2 bytes are for respectively, revealing the packet format more clearly, and making it easier to go back and identify in the future.  
Likewise below.